src/audio/module_adapter/module_adapter.c

```diff
@@ -1107,6 +1114,8 @@ static int module_adapter_sink_source_copy(struct comp_dev *dev)
 
 	comp_dbg(dev, "module_adapter_sink_source_copy(): start");
 
+	module_adapter_sink_src_update(dev, mod);
```

Comment:
Why you want to get new set of sink/src pointers before each copy? 
What is "dynamic multi-source issue"?

Reply:
This commit does fix the CI issue reported for https://github.com/thesofproject/sof/pull/8537 PR. Verified on draft PR https://github.com/thesofproject/sof/pull/8576 with https://github.com/thesofproject/sof/pull/8580/commits/1855581b2353403caaa307e9597659751e316f46 commit included.

Here is an example use case that requires this commit: consider two pipelines, one with mixin and the other with mixout, mixin is connected to mixout. After pipelines started, .prepare() is called for mixin and mixout. So now mixout has num_of_sources set to 1. After some time driver creates third pipeline containing mixin, binds that mixin to mixout and starts this new pipeline. .prepare() will be called for this new mixin, however, .prepare() will not be called for mixout!!! And mixout's num_of_sources is still 1, which is now wrong. This is because mixout pipeline was already running. To call .prepare() for mixout, its pipeline should be paused, then reset and then started again. However, driver does not do this.

So this commit fixes the problem. However, you are right, it is excessive to update sinks/sources at every copy call for each module. It would be better to move the module_adapter_sink_src_update() to bind() and unbind() handlers.

Reply:
sure, I agree to move it to bind & unbind. Actually I did it in bind & unbind. Let's change it.

Reply:
for smart amp, the reference buffer from capture will attach to playback stream, so the source count will be changed. This is the issue